backend/controller/user.go

```diff
@@ -2,14 +2,87 @@ package controller
 
 import (
 	"api/config"
+	"api/helpers"
 	"api/models"
 	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"log"
 	"net/http"
+	"strconv"
 )
 
 func GetAllUsers(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
 	users := []models.User{}
-	config.DB.Find(&users)
+	result := config.DB.Find(&users)
+	if result.Error != nil {
+		helpers.DBErrorHandling(result.Error, w, r)
+		return
+	}
+	//go yells at you if you don't error check this, better to be safe
+	errJson := json.NewEncoder(w).Encode(&users)
+	if errJson != nil {
+		http.Error(w, "500", http.StatusInternalServerError)
+		return
+	}
```

Comment:
how is data returned?

Reply:
its not in these lines, but line 21, when we try to write &users, that is what is returning the data